### ISSUE TITLE
Provide deoplete omni patterns as a variable

### ISSUE THREAD

**Issue by languitar** [lervag/vimtex#885]
I just noted that the my version of `g:deoplete#omni#input_patterns.tex` was outdated and the documentation provided a newer version with more completion options. Couldn't vimtex provide the documented version in some variable that on can assign to `g:deoplete#omni#input_patterns.tex`? That way I would have caught the changes automatically without accidentally noticing them.

**Comment by lervag** [comment]
Yes, this is a very good idea. Thanks!

**Comment by lervag** [comment]
Let me know what you think about this, in particular the update to the docs.

**Comment by languitar** [comment]
Seems this conflicts with lazy plugin loading in vim-plug. I'll have to check how to enable that.

**Comment by languitar** [comment]
Hm, I can't see any hook that would allow me to perform something in case a plugin gets loaded. So maybe this would need to be turned around that vimtex defines these things (given some option). I don't see how this would work with lazy loading then. :/

**Comment by lervag** [comment]
There is no reason to lazily load vimtex. Vimtex is a filetype plugin that uses the autoload feature, and so the only script that is actually sourced before opening a tex file is the `ftplugin/tex.vim`, and this does not impact loading times. Thus my suggestion is that you load vimtex without restricting it to tex files. And in this case, there should be no conflict.

**Comment by languitar** [comment]
Ok, makes sense. Without lazy loading things work as expected. Thanks!

**Comment by Shougo** [comment]
vim-tex plugin does not have plugin/ files.  So it is true.

**Comment by lervag** [comment]
No problem, @languitar. And thanks for commenting, @Shougo. I've updated my previous comment, where I stated that "the only script that is actually sourced before opening a tex file is the ftplugin/tex.vim, and this does not impact loading times". This is wrong. The `ftplugin` is only sourced when loading a LaTeX buffer. Thus vimtex has no impact on loading times except when opening a buffer with `filetype=tex`.
